crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,234 @@
+function CrowdsourceHinter(runtime, element, data){
+    //switching units back to a previous question will make a second hinter start up.
+    //executeHinter is used to disable the hinter after switching units in an edX course.
+    var executeHinter = true;
+    $(".crowdsourcehinter_block", element).hide();
+
+    if(executeHinter){
+    var isShowingHintFeedback = false;
+    var voted = false;
+    $(".csh_HintsToUse", element).text("");
+
+    function stopScript(){
+    //This function is used to prevent a particular instance of the hinter from acting after
+    //switching between edX course's units. 
+        executeHinter = false;
+    }
+    Logger.listen('seq_next', null, stopScript);
+    Logger.listen('seq_prev', null, stopScript);
+    Logger.listen('seq_goto', null, stopScript);
+
+    //data about the problem obtained from Logger.listen('problem_graded') is passed on to the onStudentSubmission.
+    //directly passing data to onStudentSubmission does not appear to work
+    function get_event_data(event_type, data, element){
+        onStudentSubmission(data);
+    }
+    Logger.listen('problem_graded', data.hinting_element, get_event_data);
+
+    function onStudentSubmission(problem_graded_event_data){
+    //This function will determine whether or not the student correctly answered the question.
+    //If it was correctly answered it will begin the process for giving feedback on hints.
+        if (problem_graded_event_data[1].search(/class="correct/) === -1){
```

Comment:
Here, shorter comment would be more readable. 

        // Called on student answer submission. 
        // Either show hint, or allow student to improve hint database. 
        if (problem_graded_event_data[1].search(/class="correct/) === -1){ // correct answer submitted